src/main/java/org/template/recommendation/indexeddataset/BiMapJava.java

```diff
@@ -0,0 +1,83 @@
+package org.template.recommendation.indexeddataset;
+
+import java.util.HashMap;
+import org.apache.mahout.math.indexeddataset.BiMap;
+import scala.collection.JavaConverters;
+import scala.collection.immutable.Map;
+import scala.collection.JavaConverters$;
+
+/**
+ * Created by Alvin Zhu on 2/27/17.
+ * Mini-wrapper for org.apache.mahout.math.indexeddataset.Bimap
+ *
+ */
+/**
+ * Immutable Bi-directional Map.
+ * @param m Map to use for forward reference
+ * @param i optional reverse map of value to key, will create one lazily if none is provided
+ *          and is required to have no duplicate reverse mappings.
+ */
+public class BiMapJava<K,V> {
+    BiMap<K,V> bmap;
```

Comment:
I think protected here is fine. This is a helper data structure for IndexedDataset, and basically won't be accessed outside the package.